confgenerator/built-in-config-linux.yaml

```diff
@@ -13,7 +13,7 @@ metrics:
   receivers:
     hostmetrics:
       type: hostmetrics
-      collection_interval: 60s
+      collection_interval: 1m0s
```

Comment:
Bleh. Is there no way we can force this to be serialized as seconds? These are user-visible sample configs, so it is not a great idea to change them…

Looks like the goccy/go-yaml library [does not support this](https://github.com/goccy/go-yaml/blob/master/encode.go#L581), but maybe a custom marshaler?

Reply:
ISTR last I looked into this it was not possible to implement a custom marshaler (hence why I had to fix it upstream in the first place). I think we just have to live with this; this is the canonical serialization of a `time.Duration(60000000000)` in Go.

Reply:
I recall a GitHub issue in the golang/go repo where custom `time.Duration` string formatting was discussed, and it was denied because of backwards compatibility. I'm going to keep looking for the issue, I wish I'd saved it. But the outcome of that was that we'll have to have some custom function to format this as `60s` if we want it that way.

The last intrusive way I can think to do that would be to keep the `CollectionIntervalString` function, check if the value is 60 seconds, and deliberately output `60s` in that scenario, otherwise just return the string formatted duration.

Reply:
I think we just need to live with it; this is the canonical serialization of 60s.